wurst/util/Illusion.wurst

```diff
@@ -0,0 +1,102 @@
+package Illusion
+// Credits goes to Flux for original vJass system, Trokkin for Wurst transition
+import NoWurst
+import Trigger
+import Unit
+import Orders
+import HashMap
+import DamageModify
+import Printing
+import OnUnitEnterLeave
+import AbilityObjEditing
+import InstantDummyCaster
+
+@configurable constant int ILLUSION_ID = 'AILS'
+@configurable constant int REFRESH_COUNT = 30
+
+unit lastCreatedIllusion
+trigger illusionCatchTrigger
+
+function unit.getIllusion(int abilId, int lvl) returns unit
+	unit r = null
+	if this != null and this.isAlive()
+		lastCreatedIllusion = null
+		illusionCatchTrigger.enable()
+		if InstantDummyCaster.castTarget(null, abilId, lvl, Orders.wandOfIllusion, this)
+			if lastCreatedIllusion != null
+				r = lastCreatedIllusion
+			else
+				Log.error("[Illusion] No illusion created (or found)")
+		else
+			Log.error("[Illusion] Issued illusion create order failed")
+		illusionCatchTrigger.disable()
+	else
+		Log.error("[Illusion] Source unit dead or non-existing")
+	return r
+
+init
+	illusionCatchTrigger = CreateTrigger()
+		..registerAnyUnitEvent(EVENT_PLAYER_UNIT_SUMMON)
+		..addCondition(Condition(() -> begin
+			if IsUnitIllusion(GetSummonedUnit())
+				lastCreatedIllusion = GetSummonedUnit()
+		end))
+		..disable()
+
+	onLeave(() -> begin
+		let ill = Illusion.get(getEnterLeaveUnit())
+		if(ill != null)
+			destroy ill
+	end)
+	addDamageModifier() d ->
```

Comment:
I'm confused. Does this syntax work or not? I think this wanted to be `addDamageModifier(d -> ... )`?